apps/src/applab/firebaseStorage.js

```diff
@@ -398,4 +400,82 @@ FirebaseStorage.renameColumn = function (tableName, oldName, newName, onSuccess,
     .then(onSuccess, onError);
 };
 
+/**
+ * Parses a CSV string into records data in a format that can be written into the
+ * records node of a firebase table.
+ * @param {string} csvData
+ * @returns {Promise} A promise containing recordsData or an error message.
+ */
+function parseRecordsDataFromCsv(csvData) {
+  return new Promise((resolve, reject) => {
+    parseCsv(csvData, {columns: true}, (error, records) => {
+      if (error) {
+        return reject(error);
+      }
+      resolve(records);
+    });
+  }).then(records => {
+    let recordsData = {};
+    records.forEach((record, index) => {
+      const id = index + 1;
+      Object.keys(record).forEach(key => {
+        record[key] = castValue(record[key]);
+      });
```

Comment:
I think [`_.mapValues`](https://lodash.com/docs#mapValues) would also be appropriate here:
```javascript
_.mapValues(record, castValue);
```